bowling/Game.cpp

```diff
@@ -0,0 +1,203 @@
+    int frameCount = 1;
+    std::getline(file, rolls);
+    for (int currentRoll = 0; currentRoll < rolls.size(); ++currentRoll) {
+        if (isStrike(rolls[currentRoll])) {
+            currentFrame = Frame('X', ' ');
+            currentRoll++;
+            if (isLastFrame(frameCount)) {
+                currentRoll++;
+            }
+        } else if (isSpare(rolls[currentRoll + 1])) {
+            currentFrame = Frame(rolls[currentRoll], '/');
+            currentRoll += 2;
+            if (isLastFrame(frameCount)) {
+                currentRoll++;
+            }
+        } else {
+            if (currentRoll == rolls.size() - 1) {
+                currentFrame = Frame(rolls[currentRoll], ' ');
+            } else {
+                currentFrame = Frame(rolls[currentRoll], rolls[currentRoll + 1]);
+            }
+            currentRoll += 2;
+        }
+        frameCount++;
+        playerRolls.push_back(currentFrame);
+    }
+}
+
+bool Game::isLastFrame(int frameCount) const {
+    return frameCount == 10;
+}
+
+bool Game::isStrike(char currentRoll) const {
+    return currentRoll == 'X';
+}
+
+bool Game::isStrike(const Frame& frame) const {
+    return frame.getFirstRoll() == 'X' ? true : false;
+}
+
+bool Game::isSpare(char nextRoll) const {
+    return nextRoll == '/';
+}
+
+bool Game::isSpare(const Frame& frame) const {
+    return frame.getSecondRoll() == '/' ? true : false;
+}
+
+Game::Status Game::getGameStatus() const {
+    if (players.size() == 0) {
+        return Game::Status::NO_GAME;
+    }
+    if (std::any_of(players.begin(), players.end(), [](const PlayerData& player) {
+            return !(player.rolls.size() == 10 || player.rolls.size() == 11);
+        })) {
+        return Game::Status::IN_PROGRESS;
+    }
+    if (std::all_of(players.begin(), players.end(), [](const PlayerData& player) {
+            return player.rolls.size() == 10 || player.rolls.size() == 11;
+        })) {
+        return Game::Status::FINISHED;
+    }
+    return Game::Status::NO_GAME;
+}
+
+std::vector<Frame> Game::conversionCharNumbersToInt(const std::vector<Frame>& rolls) const {
+    std::vector<Frame> convertedRolls{};
+    Frame currentFrame{};
+    char conversionNumber = '0';
+    size_t firstRoll{};
+    size_t secondRoll{};
+    for (size_t i = 0; i < rolls.size(); i++) {
+        firstRoll = rolls[i].getFirstRoll();
+        secondRoll = rolls[i].getSecondRoll();
+        if (!isStrike(rolls[i]) && !isSpare(rolls[i])) {
+            firstRoll = (rolls[i].getFirstRoll() - conversionNumber);
+            secondRoll = (rolls[i].getSecondRoll() - conversionNumber);
+        }
+        if (isSpare(rolls[i])) {
+            firstRoll = (rolls[i].getFirstRoll() - conversionNumber);
+        }
+        if (isBadCharacter(rolls[i].getFirstRoll())) {
+            firstRoll = 0;
+        }
+        if (isBadCharacter(rolls[i].getSecondRoll())) {
+            secondRoll = 0;
+        }
+        currentFrame = (Frame(firstRoll, secondRoll));
```

Comment:
you try to keep your integer numbers in Frame, which keep 'char'. They will have non-printable characters. Probably a new type for them would be better - just size_t not char